src/schema/builder.rs

```diff
@@ -375,14 +394,14 @@ impl SchemaBuilder {
         MapBuilder::name(name)
     }
 
-    pub fn build(self, root: NameRef) -> Result<Schema, Error> {
+    pub fn build(&self, root: NameRef) -> Result<Schema, Error> {
         self.validate(&root)?;
         Ok(Schema {
-            namespace_names: self.namespace_names,
-            type_names: self.type_names,
-            aliases: self.aliases,
-            reverse_aliases: self.reverse_aliases,
-            types: self.types,
+            namespace_names: self.namespace_names.clone(),
```

Comment:
Humm this was done to avoid doing lots of `memcpy` for schema building, if this is needed the whole approach is probably bust for performance.